assets/js/googlesitekit/widgets/components/WidgetAreaRenderer.test.js

```diff
@@ -121,6 +128,116 @@ describe( 'WidgetAreaRenderer', () => {
 		} );
 	} );
 
+	it( 'should only render widgets the user has access to in a view-only viewContext', async () => {
+		createWidgets( registry, areaName, [
+			{
+				Component: WidgetComponent,
+				modules: 'search-console',
+				slug: 'one',
+				width: WIDGET_WIDTHS.FULL,
+			},
+			{
+				Component: WidgetComponent,
+				modules: 'search-console',
+				slug: 'two',
+				width: WIDGET_WIDTHS.FULL,
+			},
+			{
+				Component: () => <div>AdSense is here</div>,
+				modules: 'adsense',
+				slug: 'three',
+				width: WIDGET_WIDTHS.FULL,
+			},
+		] );
+
+		// Add our dashboard sharing capabilities into state.
+		fetchMock.getOnce(
+			/^\/google-site-kit\/v1\/core\/modules\/data\/list/,
+			{
+				body: FIXTURES,
+				status: 200,
+			}
+		);
+		const capabilitiesBaseVar = '_googlesitekitUserData';
+		const capabilitiesWithPermission = {
+			permissions: {
+				googlesitekit_view_dashboard: true,
+				googlesitekit_manage_options: true,
+				'googlesitekit_read_shared_module_data::["adsense"]': false,
+				'googlesitekit_read_shared_module_data::["search-console"]': true,
+			},
+		};
+		global[ capabilitiesBaseVar ] = capabilitiesWithPermission;
+
+		const widgets = registry.select( CORE_WIDGETS ).getWidgets( areaName );
+		const { container } = render(
+			<WidgetAreaRenderer slug={ areaName } />,
+			{ registry, viewContext: VIEW_CONTEXT_DASHBOARD_VIEW_ONLY }
+		);
+
+		await waitFor( () => {
```

Comment:
Is this only to avoid errors from state updating after render? If so, I think we just need to `await waitForRegistry()` instead.

Reply:
Actually none of it seems needed, but this was just copied from a neighbouring test. Seems fine not to use at all though, so I've removed it 👍🏻